eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -290,6 +289,9 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
           case (count, (msg, origins)) if !timestampInRange(msg, origins, d.gossipTimestampFilter) =>
             // the peer has set up a filter on timestamp and this message is out of range
             count
+          case (count, (u: ChannelUpdate, _)) if u.isPrivate =>
+            log.error("private channel update for scid={} was incorrectly added to staggered broadcast", u.shortChannelId)
```

Comment:
So this is the actual improvement brought by the spec change. Less nice that strong typing all the way down, but still very useful as runtime debugging.